UnityProject/Assets/Scripts/Objects/ExecutionDevices/ExecutionDeviceController.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Globalization;
+using AdminTools;
+using UnityEngine;
+
+namespace Objects.ExecutionDevices
+{
+	public class ExecutionDeviceController : MonoBehaviour, IRightClickable
+	{
+		private IExecutionDevice device;
+		public GameObject Victim { get; set; }
+
+		private void Awake()
+		{
+			device ??= GetComponent<IExecutionDevice>();
+		}
+
+		public void Execute(GameObject executioner = null)
+		{
+			if (Victim == null)
+			{
+				if(executioner != null) Chat.AddExamineMsg(executioner, "There's nothing to execute!");
+				return;
+			}
+			StartCoroutine(device.ExecuteTarget());
+			LogExecution(executioner);
+		}
+
+		public void ReleaseVictim()
+		{
+			if (device == null)
+			{
+				Logger.LogError($"[ExecutionDeviceController/ReleaseVictim] - There's no device interface on {gameObject.name}!");
+				return;
+			}
+			device.OnLeaveDevice(Victim);
+		}
+
+		private void LogExecution(GameObject executioner)
+		{
+			if (Victim == null || executioner == null) return;
+			var time = DateTime.Now.ToString(CultureInfo.InvariantCulture);
+			UIManager.Instance.playerAlerts.ServerAddNewEntry(time, PlayerAlertTypes.RDM, executioner.Player(),
+				$"{time} : {executioner.Player().Mind.CurrentPlayScript.playerName} attempted to execute {Victim.ExpensiveName()} at {Victim.AssumedWorldPosServer()}.");
+		}
+
+		public RightClickableResult GenerateRightClickOptions()
+		{
+			var newList = new RightClickableResult();
+			newList.AddElement("Execute", () => Execute(PlayerManager.LocalPlayerObject));
+			newList.AddElement("Release Victim", ReleaseVictim);
+			return newList;
```

Comment:
I'm wondering if it should post it on common?
and when you execute someone you have to put in the reason,
I just imagine walking around the station and you hear "Captain Ballymena Byblos was executed by guillotine for" not giving the clown shoes back to the clown""

Reply:
Seems unrelated to this PR. Maybe it can be a module that people can install on the future but for now this doesn't make sense for a wooden piece of equipment that is labeled under the "tribal" section

Reply:
ok